--- a/p2c-2.01/src/Makefile
+++ b/p2c-2.01/src/Makefile
@@ -38,6 +38,8 @@ MANFILE = p2c.cat          # human-readable manual (for cat.1)
 
 # Compiler options
 CC = cc                    # you may wish to use gcc here instead
+RANLIB = ranlib
+AR = ar
 OPT = # -O		   # uncomment this for optimization
 DEB = # -g		   # uncomment this for debugging
 DEFS =	-DTEST_MALLOC      # place other -D types of things here # TDS 2015nov18
@@ -102,7 +104,7 @@ makeproto: makeproto.c
 
 # Making the p2c runtime library
 libp2c.a: $(LIBOBJS)
-	ar r libp2c.a $(LIBOBJS) $(OTHERLIBOBJS)
+	$(AR) r libp2c.a $(LIBOBJS) $(OTHERLIBOBJS)
 
 p2clib.o: p2clib.c
 	$(CC) -c $(CFLAGS) p2clib.c
@@ -175,7 +177,7 @@ $(BINDIR)/p2cc: p2cc
 
 $(LIBDIR)/libp2c.a: libp2c.a
 	$(COPY)  libp2c.a     $(LIBDIR)/libp2c.a
-	if [ -f /usr/bin/ranlib -o -f /bin/ranlib ]; then ranlib $(LIBDIR)/libp2c.a; fi
+	$(RANLIB) $(LIBDIR)/libp2c.a
 
 $(MANDIR)/p2c.1: $(MANFILE)
 	$(COPY)  $(MANFILE)   $(MANDIR)/p2c.1
